### DISCUSSION TITLE
sq-001 and stm32f101

### DISCUSSION THREAD

**Discussion by bm16ton** [Ralim/IronOS#1386] [General]
Hello, so i used iron-os on my sq-001 about a year ago and whenni started with sensor errors etc, during debug i rreverted to stock firmware (wasnt firmware related the metal rings just come loose a lot). i again today had to open and reseat/tighten everything and for first time noticed its running a stm32f101t8u6. Dunno how i missed tthat before. I did a very quick look around ironos source and only see f103 startup code...is the f103 the only officially supported mcu target? Tge f101 datasheet doesnt mention usb (st always hidding extra features usb/ram/storage) and ssays max fcpu is 36mhz so i guess i was overclocking the lil fella when running ironos? Whats the clock rate for the f103 set to in ironos? i assume 48mhz or 72mhz. Anywho just curious.

**Comment by Ralim** [comment]
Hia,
The SQ001 is not supported as they have a tendency to fit whatever chip they can buy.
So yeah, technically the f101 is the "f103 that might have failed qc", but generally they have all the parts of the f103.

This means that _technically_ USB is out of spec, as is the clock rate of 72MHz, and I think some of the internal clocks are also well out of spec. 

But it _generally_ works for people.

The code is running assuming its an F103, which generally works due to the older F101's often just being F103's.

So if it works for you; great 😁 Just I cant really provide support for odd quirks as much

**Comment by bm16ton** [comment]
awesome! Thank you for tthe info! I was just curious, luckily dont need any
support but am glad to know ironos runs at 72mhz and this guy whos supposed
to run at 36mhz was keeping up. unfortunatly i think itd be a bad idea to
run ironos on it unless i find out the stock firmware is also running 72mhz
then i guess it wouldnt matter. Between the chips failing sum unknown test
or market demands requiring cheaper less capable chips out of the handful
of stm32 variants i have a very few dont have much larger storage/ram but
this is so far my first that has usb. hrm i have a f410 sumwhere that
claims no usb ill hafta look into that. But again thank you for the info!

**Reply by Ralim** [reply]
Stock OS is 64MHz last I checked so its also an overclock ^_^
